support for react version 0.11.0

skratchdot opened the pull request:
React v0.11.0 was recently released so the specified peerDependency in rrouter v0.4.0 is failing:

http://facebook.github.io/react/blog/2014/07/17/react-v0.11.html

The only "listed" breaking change is to getDefaultProps(), but you only appear to be using that in your PathnameRouting-test.js file- and it seems like the react v0.11.0 change won't break that instance.

I've tested a local app and things seem to be working with this package.json change.

NOTE: I've also updated the version number to v0.5.0 so you'll have to republish for this change to really take effect.

skratchdot committed:
support for react version 0.11.0
React v0.11.0 was recently released so the specified peerDependency in rrouter v0.4.0 is failing:

http://facebook.github.io/react/blog/2014/07/17/react-v0.11.html

The only listed breaking change is to getDefaultProps(), but you only appear to be using that in your PathnameRouting-test.js file- and it seems like the react v0.11.0 change won't break that instance.

I've tested a local app and things seem to be working with this package.json change.

NOTE: I originally updated package.json to v0.5.0, but after your comment in the pull request, I kept it at v0.4.0

jnak commented:
lgtm

ninjabiscuit commented:
:+1:

sketchpixy commented:
:+1:

skratchdot commented:
@andreypopp - Let me know if there's anything I should do to clean this up, or if you have any concerns about bumping the version.

andreypopp commented:
@skratchdot I don't think we should do `>= 0.10.0` cause React can has breaking changes before `1.0.0`. The constraint could be `>=0.10.0,<0.12.0`.

Another concern is that version in master has a lot of changes from the last release and I simply don't know if is is tested enough.

skratchdot commented:
@andreypopp - that makes sense. I went ahead and removed the v0.5.0 from the package.json so you can update whenever. I also updated the react versions using the range syntax from npm's semver: https://www.npmjs.org/doc/misc/semver.html#ranges

I tested it locally, and have been using it w/ v0.11 for a week now without issues.

simenbrekken commented:
@andreypopp Could you merge this and publish to NPM please? React 0.10 is sooo spring 2014.

andreypopp commented:
@sbrekken will do it this evening
